src/openrct2/actions/TrackDesignAction.cpp

```diff
@@ -259,7 +259,7 @@ GameActions::Result TrackDesignAction::Execute() const
 
     for (size_t i = 0; i <= MAX_VEHICLES_PER_RIDE; i++)
     {
-        auto tdIndex = std::min(i, std::size(_td.vehicle_colours) - 1);
+        auto tdIndex = i % (std::size(_td.vehicle_colours) - 1);
```

Comment:
The modulo is also not right. You don't want to exclude the last one, note that 32 % 32 = 0 and not 32

Reply:
That's true, thanks.